### ISSUE TITLE
CMAKE Windows OBJECT_PATH_MAX 

### ISSUE THREAD

**Issue by Unarmed1000** [android-ndk/ndk#207]
For even simple projects its quite easy to hit a OBJECT_PATH_MAX warning when using cmake on windows.

Warning example:
```
 The object file directory
        E:/Work/ProjectName10/Relevant/Name1/SomethingElse/Android/app/.externalNativeBuild/cmake/debug/armeabi/ProjectName10/SelfDocumentingNameHere/CMakeFiles/HeyNativeWindowPlatform.dir/

  has 181 characters.  The maximum full path to an object file is 250
  characters (see CMAKE_OBJECT_PATH_MAX).  Object file

    source/HeyNativeWindowPlatform/PlatformNativeWindowSystemFactory.cpp.o

  cannot be safely placed under this directory.  The build may not work
  correctly.
```  

While I agree its a long path a lot of it is due to the way cmake is handled.

So I used "E:/Work/ProjectName10/Relevant/Name1/SomethingElse/Android/" as my app root which is only 60 chars in a best case scenario I could probably cut it down to "E:/Work/ShrtGroup/Name1.SomethingElse" saving 21chars which still isn't a lot of wiggle room when I am hitting the limit already.

Out of the 181 characters around 40 is overhead (.externalNativeBuild/cmake/debug/armeabi/CMakeFiles/.dir) thats close to 20% of the total limit and almost 2x of what I could easily save.

So basically either the warning needs to go by using a larger CMAKE_OBJECT_PATH_MAX after its been determined to be safe! 
Or more realistically the cmake temporary file scheme needs to be improved a lot.

**Comment by DanAlbert** [comment]
Whoops, I thought I'd responded to this.

That path limit is actually imposed by Windows, not cmake: https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx#maxpath (it looks like there's a wiggle room of six chars here; I'm not sure why).

So, while we can't actually raise that limit, something might be able to be done about the very long paths that are generated for you. It's actually Android Studio that's generating those paths, not anything in the NDK, so could you file a bug in their tracker for that? http://tools.android.com/filing-bugs
